app/controllers/concerns/user_login.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module UserLogin
+  extend ActiveSupport::Concern
+
+  def login(group_kind: nil)
+    user = login_user(@credentials[:login], group_kind: group_kind)
+    if user&.ldap.present? && Ldap.login(user.ldap, @credentials[:password]) ||
+       user&.authenticate(@credentials[:password])
+      return login_success(user)
+    end
+    login_error 'Login oder Passwort sind nicht korrekt'
+  end
+
+  def login_error(error)
+    @error = error
+    logger.info error
+    render :new
+  end
+
+  def login_success(user)
+    session[:user_login] = user.login
+    redirect_to root_url
+  end
+
+  def find_user(login, group_kind: nil)
```

Comment:
```suggestion
  def find_users(login, group_kind: nil)
```
Bitte im Plural, weil eine Liste geliefert und erwartet wird, statt nur einem direkten Treffer. Bitte auch an den aufrufenden Stellen korrigieren.